### PR TITLE
Buildings get locked by Demolishable activity before their destruction

### PR DESCRIPTION
**Pull request by HenrytheSlav**
Solves #10846 ! Made mostly by @abcdefg30 so love (or hate) him, I only debugged the last part. :tophat:

### CODE REVIEW

File: OpenRA.Mods.Common/Activities/Demolish.cs [hunk 1]
```diff
@@ -57,6 +58,14 @@ protected override void OnInside(Actor self)
 				if (cloak != null && cloak.Info.UncloakOn.HasFlag(UncloakType.Demolish))
 					cloak.Uncloak();
 
+				if (!chargePlanted)
+				{
+				chargePlanted = true;
+				var building = target.TraitOrDefault<Building>();
+				if (building != null)
+					building.Lock();
```
Comment (reaperrr) [inline OpenRA.Mods.Common/Activities/Demolish.cs]: All four lines within the brackets need one additional tab indentation.

**Comment by HenrytheSlav** [comment]
Sorry for that, updated with correct indentation.

**Comment by obrakmann** [comment]
works as advertised, :+1:

File: OpenRA.Mods.Common/Activities/Demolish.cs [hunk 2]
```diff
@@ -57,6 +58,14 @@ protected override void OnInside(Actor self)
 				if (cloak != null && cloak.Info.UncloakOn.HasFlag(UncloakType.Demolish))
 					cloak.Uncloak();
 
+				if (!chargePlanted)
+				{
+					chargePlanted = true;
```
Comment (Mailaender) [inline OpenRA.Mods.Common/Activities/Demolish.cs]: Why is this required?

Comment (GraionDilach) [inline OpenRA.Mods.Common/Activities/Demolish.cs]: There couléd be multiple actors entering to the same target (D2k Saboteurs came to my mind) in which case multiple `Lock`s might be errandeous.

Comment (pchote) [inline OpenRA.Mods.Common/Activities/Demolish.cs]: This particular code only ever runs once, so the check does nothing.

**Comment by epicelite** [comment]
I approve!

File: OpenRA.Mods.Common/Activities/Demolish.cs (hunk 2 above)
Comment (HenrytheSlav) [inline OpenRA.Mods.Common/Activities/Demolish.cs]: Sorry I thought this runs on every tick like external capturing which I took as template. I will fix that.

**Comment by HenrytheSlav** [comment]
Rewritten, works, but idk if it is okay with coding style etc.

**Comment by GraionDilach** [comment]
:+1:

**Comment by Mailaender** [comment]
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/11e324f36332bc3e801fda533ff130e2b6fa0e89...362a05194bfbccd3f6f4dc58ee82e3b80e061706)
